src/ixdat/measurements.py

```diff
@@ -0,0 +1,375 @@
+        new_name = self.name + " AND " + other.name
+        new_technique = self.technique + " AND " + other.technique
+
+        from .techniques import TECHNIQUE_CLASSES
+
+        if new_technique in TECHNIQUE_CLASSES:
+            cls = TECHNIQUE_CLASSES[new_technique]
+        elif self.__class__ is other.__class__:
+            cls = self.__class__
+        else:
+            cls = Measurement
+
+        new_series_list = self.series_list
+        new_series_list.append(other.series_list)
+        new_component_measurements = self.component_measurements or [self]
+        new_component_measurements.append(other.component_measurements or [other])
+        obj_as_dict.update(
+            name=new_name,
+            series_list=new_series_list,
+            component_measurements=new_component_measurements,
+        )
+        return cls.from_dict(obj_as_dict)
+
+
+#  ------- Now come a few module-level functions for series manipulation ---------
+#  TODO: move to a .build module or similar
+
+
+def append_series(series_list):
+    """Return series appending series_list relative to series_list[0].tseries.tstamp"""
+    s0 = series_list[0]
+    if isinstance(s0, TimeSeries):
+        return append_tseries(series_list)
+    elif isinstance(s0, TimeSeries):
+        return append_tseries(series_list)
+    raise BuildError(
+        f"An algorithm of append_series for series like {s0} is not yet implemented"
+    )
+
+
+def append_vseries_by_time(series_list):
+    """Return new series with the data in series_list appended"""
+    name = series_list[0].name
+    cls = series_list[0].__class__
+    unit = series_list[0].unit
+    data = np.array([])
+
+    for s in series_list:
+        if not (s.name == name and s.unit == unit and s.__class__ == cls):
+            raise BuildError(f"can't append {series_list}")
+        data = np.append(data, s.data)
+
+    tseries = append_tseries([s.tseries for s in series_list])
+    vseries = cls(name=name, unit=unit, data=data, tseries=tseries)
+    return vseries
+
+
+def append_tseries(series_list, tstamp=None):
+    """Return new TSeries with the data appended relative to series_list[0].tstamp"""
+    name = series_list[0].name
+    cls = series_list[0].__class__
+    unit = series_list[0].unit
+    tstamp = tstamp or series_list[0].tstamp
+    data = np.array([])
+
+    for s in series_list:
+        if not (s.name == name and s.unit == unit and s.__class__ == cls):
+            raise BuildError(f"can't append {series_list}")
+        data = np.append(data, s.data + s.tstamp - tstamp)
+
+    tseries = cls(name=name, unit=unit, data=data, tstamp=tstamp)
+    return tseries
+
+
+def fill_object_list(object_list, obj_ids, cls=None):
+    """Add PlaceHolderObjects to object_list for any unrepresented obj_ids"""
+    cls = cls or object_list[0].__class__
+    object_list = object_list or []
+    provided_series_ids = [s.id for s in object_list]
+    if not obj_ids:
+        return
```

Comment:
None or object_list return behavior not documentet

Reply:
Updated docstring:
```
    """Add PlaceHolderObjects to object_list for any unrepresented obj_ids.

    Args:
        object_list (list of objects or None): The objects already known,
            in a list. This is the list to be appended to. If None, an empty
            list will be appended to.
        obj_ids (list of ints or None): The id's of objects to ensure are in
            the list. Any id in obj_ids not already represented in object_list
            is added to the list as a PlaceHolderObject
        cls (Saveable class): the class remembered by any PlaceHolderObjects
            added to the object_list, so that eventually the right object will
            be loaded.
    """
```

Reply:
What I meant was actually the return value, since it may be an object_list or None, which might be good to mention, even though the return value may be secondary

Reply:
I don't know what you mean here. It always returns `object_list`. `object_list` can be an empty list if all inputs are empty, but it can't be `None`.

Reply:
Ah. I don't know if the code got updated, but the version quoted above has an empty return:
```python
if not obj_ids:
    return
```

Reply:
Oh, sorry, it did indeed. That has become:
```
    if not obj_ids:
        return object_list
```
... wish github included the updated code with these threads but I guess that would be challenging to implement.